### PR TITLE
CI: enforce black style

### PR DESCRIPTION
**Pull request by sappelhoff**
cc @a-hurst this PR's CI should fail now (only the first commit is expected to fail)

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -69,6 +69,7 @@ exclude =
   examples/
 
 [isort]
+profile=black
```
Comment (sappelhoff) [inline setup.cfg:72]: this is another important line to make isort (for auto formatting imports) work nicely with black (for auto formatting everything else)

**Comment by sappelhoff** [comment]
you said you were using pylint, so this may be of interest to you: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html?highlight=isort#pylint

File: .github/workflows/python_tests.yml [hunk 2]
```diff
@@ -38,7 +38,8 @@ jobs:
       run: |
         flake8 --docstring-convention numpy .
         check-manifest .
-        black .
+        black . --check
+        isort . --check
```
Comment (sappelhoff) [inline .github/workflows/python_tests.yml:42]: and this one doesn't hurt to add either

**Review by a-hurst — APPROVED** [review]
Looks good to me! Honestly, apart from my hefty use of single-quotes and the missing commas following final elements in lists, I'm amazed at how well we've followed strict Black style without being consciously aware of it.

I'll set up Black with pylint so I can stay consistent with this style going forwards.

**Pull request merged**